controllers/webhook/ksvc_validator.go

```diff
@@ -0,0 +1,113 @@
+			return nil
+		}
+	}
+
+	// Only validate the KSVC if it is owned by KServe controller
+	ksvcMetadata := ksvc.GetObjectMeta()
+	if ksvcMetadata == nil {
+		log.V(1).Info("Skipping validation of Knative Service because it does not have metadata")
+		return nil
+	}
+	ksvcOwnerReferences := ksvcMetadata.GetOwnerReferences()
+	if ksvcOwnerReferences == nil {
+		log.V(1).Info("Skipping validation of Knative Service because it does not have owner references")
+		return nil
+	}
+	isOwnedByKServe := false
+	for _, owner := range ksvcOwnerReferences {
+		if owner.Kind == constants.InferenceServiceKind {
+			if strings.Contains(owner.APIVersion, kservev1beta1.SchemeGroupVersion.Group) {
+				isOwnedByKServe = true
+			}
+		}
+	}
+	if !isOwnedByKServe {
+		log.V(1).Info("Skipping validation of Knative Service because it is not owned by KServe")
+		return nil
+	}
+
+	// Since the Ksvc is owned by an InferenceService, it is known that it is required to be
+	// in the Mesh. Thus, the involved namespace needs to be enrolled in the mesh.
+	// Go and check the ServiceMeshMemberRoll to verify that the namespace is already a
+	// member. If it is still not a member, reject creation of the Ksvc to prevent
+	// creation of a Pod that would not be in the Mesh.
+	smmrQuerier := resources.NewServiceMeshMemberRole(v.client)
+	smmr, fetchSmmrErr := smmrQuerier.FetchSMMR(ctx, log, types2.NamespacedName{Name: constants.ServiceMeshMemberRollName, Namespace: constants.IstioNamespace})
+	if fetchSmmrErr != nil {
+		log.Error(fetchSmmrErr, "Error when fetching ServiceMeshMemberRoll", "smmr.namespace", constants.IstioNamespace, "smmr.name", constants.ServiceMeshMemberRollName)
+		return fetchSmmrErr
+	}
+	if smmr == nil {
+		log.Info("Rejecting Knative service because the ServiceMeshMemberRoll does not exist", "smmr.namespace", constants.IstioNamespace, "smmr.name", constants.ServiceMeshMemberRollName)
+		return fmt.Errorf("rejecting creation of Knative service %s on namespace %s because the ServiceMeshMemberRoll does not exist", ksvc.Name, ksvc.Namespace)
+	}
+
+	log = log.WithValues("smmr.namespace", smmr.Namespace, "smmr.name", smmr.Name)
+
+	if smmr.Status.ConfiguredMembers == nil {
+		log.Info("Rejecting Knative service because the service mesh does not have any namespaces")
+		return fmt.Errorf("rejecting creation of Knative service %s on namespace %s because the namespace is not a configured member of the service mesh", ksvc.Name, ksvc.Namespace)
+	}
+	for _, memberNamespace := range smmr.Status.ConfiguredMembers {
+		if memberNamespace == ksvc.Namespace {
+			log.V(1).Info("The Knative service is accepted")
+			return nil
+		}
+	}
+
+	log.Info("Rejecting Knative service because its namespace is not a member of the service mesh")
+	return fmt.Errorf("rejecting creation of Knative service %s on namespace %s because the namespace is not a configured member of the mesh", ksvc.Name, ksvc.Namespace)
```

Comment:
Not sure you really need a log here - the error you are returning seems to be sufficient and I assume it will be logged somehow?

In addition the info message is misleading. It can be pending or failing if I'm reading this right.

Reply:
The error is not written to the controller logs. The error is returned to the client that tried to create the resource. So, the client chooses what to do with the message. Because of this, I chose to add the log message to make it clear what is happening in odh-model-controller side.

About `pending` or `failing`, I'm not sure if it is worth to go that deep here :-) Stating it is _not configured_ should is enough for me to make the user go and diagnose it, in case it stays on one of those states for longer than expected.

Reply:
Ok, I thought this is handled by operator-sdk, but if that is not the case logging is helpful here.

> About pending or failing, I'm not sure if it is worth to go that deep here :-) Stating it is not configured is enough for me to make the user go and diagnose it, in case it stays on one of those states for longer than expected.

I was coming from the angle that "not configured" is a bit ambiguous here. Does it mean that someone forgot to put the proper label on the namespace so it is not really seen by the mesh OR it cannot be configured by the mesh because... other errors? But I agree it might be going to deep.